### ISSUE TITLE
Turret vehicle on the Hold Fire Stance bug

### ISSUE THREAD

**Issue by UppsAlert** [OpenRA/OpenRA#21674]
The unit with a Turret has to be on the Hold Fire Stance, 
the units ignore new Attack move orders they seem to be locked on actors they attacked before even when canceled with a normal move order. 

[Hold Fire Stance bug.zip](https://github.com/user-attachments/files/18126497/Hold.Fire.Stance.bug.zip)

**Comment by PunkPun** [comment]
I dunno if this is a bug or not. I think it's more a feature that you can use hold fire for the unit to avoid retargetting something different

**Comment by UppsAlert** [comment]
That makes no sence the units stop moving at all.
The correct behavoir would be attack in vision and range. if one of the first points dosnt match then move further until vision and attack range fits. 

The units are stuck dont move and dont retargeting even with new orders.

Its a real problem in CA because they have stealth tanks with turrets which are on hold fire.

**Comment by PunkPun** [comment]
the unit responds to move orders, and stops targetting with stop command. In regards to attack move, this is very deliberate behaviour, we don't want to change position of we are already in range of a target.

The only possibly unwanted behaviour, is the unit remembering the old target after moving out of range

**Comment by UppsAlert** [comment]
> The only possibly unwanted behaviour, is the unit remembering the old target after moving out of range

Yes thats what happens, but the thing is you can give move orders and then afterwards a attack move order and they will still stand still again. It has probably to do something with "forget" about the old target.

**Comment by UppsAlert** [comment]
The dead of the enemy actor rewrites this issue and the freezed units will move again @PunkPun
